index.js

```diff
@@ -1,302 +1,289 @@
-          if (options.logFunc) {
-            options.logFunc(action, resp);
-          } else {
-            console.log(`${action}:${JSON.stringify(resp, null, 1)}`);
-          }
-        }
-      };
-
-      if (textWasPopulated) {
-        promises.push(convoFunc(Convo.SimpleResponse({
-          text : textIsNotEmpty ? final_text : final_speech,
-          speech: speechIsNotEmpty ? final_speech : final_text
-        })));
-      }
-
-      convo._rich.forEach((richResponse) => {
-        if (richResponse.media && !richResponse.capabilities && !richResponse.send) {
-          promises.push(convoFunc(richResponse.media));
-          return;
-        }
-        if (convo.conv.surface && convo.conv.surface.capabilities.has(richResponse.capabilities)) {
-          if (!textWasPopulated && richResponse.send && (typeof richResponse.media !== "string")) {
-            promises.push(convoFunc(Convo.SimpleResponse({
-              text : richResponse.send.notification,
-              speech: richResponse.send.notification
-            })));
-          }
-          promises.push(convoFunc(richResponse.media));
-        } else if (richResponse.send && convo.conv.available && convo.conv.available.surfaces.capabilities.has(richResponse.capabilities)) {
-          promises.push(convoFunc(Convo.NewSurface(
-            {
-              context: richResponse.send.context,
-              notification: richResponse.send.notification,
-              capabilities: richResponse.capabilities
-            }
-          )));
-        }
-      });
-      if (isRunningInDialogFlow) {
-        return Promise.all(promises);
-      } else {
-        return Promise.all(promises).then(() => convo);
-      }
-
-    }
-  }
-  static prepComplete(obj, action = "close", options) {
-    if (this.isPromise(obj)) {
-      return obj.then(obj => this.prepComplete(obj, action, options));
-    } else {
-      return this.complete(obj, action, options);
-    }
-  }
-  static ask(obj, options) {
-    return this.prepComplete(obj, "ask", options);
-  }
-  static close(obj, options) {
-    return this.prepComplete(obj, "close", options);
-  }
-
-  constructor(obj) {
-    if (!obj) {
-      this.conv = Convo.mockConv();
-    } else {
-      this.conv = obj.conv ? obj.conv : obj;
-    }
-
-    this.clear();
-  }
-  clear() {
-    this._write = [];
-    this._speak = [];
-    this._rich = [];
-    return this;
-  }
-  write(message) {
-    this._write.push(message.toString());
-    return this;
-  }
-  speak(message, writeAlso = true) {
-    this._speak.push(message.toString());
-    if (writeAlso) {
-      this.write(message);
-    }
-    return this;
-  }
-  present(media, capabilities = null, send = null) {
-    this._rich.push({capabilities, send, media});
-    return this;
-  }
-  promise(func) {
-    let r = func(this);
-    if (!Convo.isPromise(r)) {
-      return Promise.resolve(r);
-    }
-    return r;
-  }
-  setAccessToken(token) {
-    if (!conv.user) {
```

Comment:
`conv`  is not defined in this scope. I assumed you meant `this.conv`

Reply:
my bad. Nice catch.